tests/pyoozie/test_tags.py

```diff
@@ -0,0 +1,293 @@
+    </shell>''') == xml_to_dict_unordered(actual)
+
+
+def test_subworkflow():
+    app_path = '/user/username/workflows/cool-flow'
+
+    actual = SubWorkflow(
+        app_path=app_path,
+        propagate_configuration=False,
+        configuration=None,
+    ).xml(indent=True)
+    assert xml_to_dict_unordered('''
+    <sub-workflow>
+        <app-path>/user/username/workflows/cool-flow</app-path>
+    </sub-workflow>
+    ''') == xml_to_dict_unordered(actual)
+
+    actual = SubWorkflow(
+        app_path=app_path,
+        propagate_configuration=True,
+        configuration=None,
+    ).xml(indent=True)
+    assert xml_to_dict_unordered('''
+    <sub-workflow>
+        <app-path>/user/username/workflows/cool-flow</app-path>
+        <propagate-configuration />
+    </sub-workflow>
+    ''') == xml_to_dict_unordered(actual)
+
+    actual = SubWorkflow(
+        app_path=app_path,
+        propagate_configuration=True,
+        configuration={
+            'job_tracker': 'a_jobtracker',
+            'name_node': 'hdfs://localhost:50070',
+        },
+    ).xml(indent=True)
+    assert xml_to_dict_unordered('''
+    <sub-workflow>
+        <app-path>/user/username/workflows/cool-flow</app-path>
+        <propagate-configuration />
+        <configuration>
+            <property>
+                <name>job_tracker</name>
+                <value>a_jobtracker</value>
+            </property>
+            <property>
+                <name>name_node</name>
+                <value>hdfs://localhost:50070</value>
+            </property>
+        </configuration>
+    </sub-workflow>
+    ''') == xml_to_dict_unordered(actual)
+
+
+def test_global_configuration():
+    configuration = {
+        'mapred.job.queue.name': '${queueName}'
+    }
+
+    actual = GlobalConfiguration(
+        job_tracker='a_jobtracker',
+        name_node='hdfs://localhost:50070',
+        job_xml_files=None,
+        configuration=None,
+    ).xml(indent=True)
+    assert xml_to_dict_unordered('''
+    <global>
+        <job-tracker>a_jobtracker</job-tracker>
+        <name-node>hdfs://localhost:50070</name-node>
+    </global>
+    ''') == xml_to_dict_unordered(actual)
+
+    actual = GlobalConfiguration(
+        job_tracker='a_jobtracker',
+        name_node='hdfs://localhost:50070',
+        job_xml_files=['/user/${wf:user()}/job.xml'],
+        configuration=configuration,
+    ).xml(indent=True)
+    assert xml_to_dict_unordered('''
+    <global>
+        <job-tracker>a_jobtracker</job-tracker>
+        <name-node>hdfs://localhost:50070</name-node>
+        <job-xml>/user/${wf:user()}/job.xml</job-xml>
+        <configuration>
+            <property>
+                <name>mapred.job.queue.name</name>
+                <value>${queueName}</value>
+            </property>
+        </configuration>
+    </global>
+    ''') == xml_to_dict_unordered(actual)
+
+
+def test_email():
+    actual = Email(
+        to='mrt@theateam.com',
```

Comment:
Please use `example.com` email addresses everywhere in tests.

While real email addresses are more fun to read, using them runs the risk of leaking and/or spamming. `example.com` is [reserved by the IANA](http://www.iana.org/domains/reserved) specifically for illustrative purposes.

(I know you're not actually _sending_ anything here, but it's a good habit because someone else might.)